Trtl 1076 upgrade terraform

davehowell opened the pull request:
Upgraded to 0.12

I ran terraform plan with some fake values and it was successful.

Note: @asavoy you wanted to change something about the `find -s` in the hash script, I am not sure of the intention so haven't done anything on that.

Commit:
Ran the 0.12upgrade script

Commit:
Use pessmimistic version pinning

Commit:
Move version to main.tf

asavoy reviewed (changes requested):
i don't think the examples folder has been upgraded? i guess it hasn't been tested with the example either?

Commit:
Add pinned version to the example

davehowell commented:
> i don't think the examples folder has been upgraded? i guess it hasn't been tested with the example either?

No I didn't see the examples, but I did test it directly with `terraform plan` and provided some fake variable values and an empty Dockerfile. Thanks for pointing out the examples.

I've run the update script on the examples folder now, and it didn't change anything apart from adding the versions.tf file. I've removed that file and added the pinned version to the examples main.tf
